dnn: build failure on gcc, invalid-pch

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => 3.4, master. after #13932
- Operating System / Platform => all
- Compiler => tested on gcc >= 4.9 on ppc64le, also on x86

##### Detailed description

```bash
cc1plus: warning: /worker/buildbot/Power9_Linux_gcc-6__opencv/build/modules/dnn/test_precomp.hpp.gch/opencv_test_dnn_Release.gch: not used because `__OPENCV_TESTS' not defined [-Winvalid-pch]
[ 50%] Building CXX object modules/imgproc/CMakeFiles/opencv_test_imgproc.dir/test/test_imgproc_umat.cpp.o
In file included from <command-line>:0:0:
/worker/buildbot/Power9_Linux_gcc-6__opencv/build/modules/dnn/test_precomp.hpp:50:27: fatal error: test_common.hpp: No such file or directory
 #include "test_common.hpp"
                           ^
compilation terminated.
make[2]: *** [modules/dnn/CMakeFiles/opencv_perf_dnn.dir/build.make:63: modules/dnn/CMakeFiles/opencv_perf_dnn.dir/test/test_common.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:6697: modules/dnn/CMakeFiles/opencv_perf_dnn.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
buildbot URI:
nightly builds: [Linux-debug](http://pullrequest.opencv.org/buildbot/builders/3_4_noOCL_noICV_noSSE-lin64-debug/builds/342/steps/compile%20debug/logs/stdio) (other tests uses ccache on Linux)
gcc4.9 : https://ocv-power.imavr.com/#/builders/5/builds/46
gcc5: https://ocv-power.imavr.com/#/builders/16/builds/291
gcc6 : https://ocv-power.imavr.com/#/builders/14/builds/18
gcc7: https://ocv-power.imavr.com/#/builders/12/builds/8


##### Steps to reproduce

build dnn module with tests enabled

--- a/modules/dnn/CMakeLists.txt
+++ b/modules/dnn/CMakeLists.txt
@@ -98,6 +98,10 @@ ocv_add_perf_tests(${INF_ENGINE_TARGET}
     FILES Src ${perf_srcs}
     FILES Include ${perf_hdrs}
 )
+set_property(
+  SOURCE "${CMAKE_CURRENT_LIST_DIR}/test/test_common.cpp"
+  PROPERTY COMPILE_DEFINITIONS "__OPENCV_TESTS=1"
+)
 
 ocv_option(${the_module}_PERF_CAFFE "Add performance tests of Caffe framework" OFF)
 ocv_option(${the_module}_PERF_CLCAFFE "Add performance tests of clCaffe framework" OFF)

